azure-pipelines.yml

```diff
@@ -51,3 +51,39 @@ jobs:
 
   steps:
   - template: ci/azure-steps.yml
+
+- job: msys2_mingw
+  pool:
+    vmImage: VS2017-Win2016
+  strategy:
+    matrix:
+      i686:
+        MSYSTEM: MINGW32
+        MSYS2_ARCH: i686
+      x86_64:
+        MSYSTEM: MINGW64
+        MSYS2_ARCH: x86_64
+  steps:
+    - script: git clone https://github.com/lazka/msys2-ci-base.git %CD:~0,2%\msys64
+      displayName: Install MSYS2
+    - script: |
+        set PATH=%CD:~0,2%\msys64\usr\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem
+        %CD:~0,2%\msys64\usr\bin\pacman --noconfirm -Syyuu
+      displayName: Update MSYS2
+    - script: |
+        set PATH=%CD:~0,2%\msys64\usr\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem
+        %CD:~0,2%\msys64\usr\bin\pacman --noconfirm --needed -S git base-devel mingw-w64-$(MSYS2_ARCH)-toolchain mingw-w64-$(MSYS2_ARCH)-python3 mingw-w64-$(MSYS2_ARCH)-python3-setuptools mingw-w64-$(MSYS2_ARCH)-python2 mercurial
+      displayName: Install Dependencies
+    - script: |
+        set PATH=C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem
+        set CHERE_INVOKING=yes
+        %CD:~0,2%\msys64\usr\bin\bash -lc "wget https://github.com/mesonbuild/cidata/raw/master/ninja.exe; mv ninja.exe /$MSYSTEM/bin"
+        set MESON_FIXED_NINJA=1
+        set PATHEXT=%PATHEXT%;.py
+        %CD:~0,2%\msys64\usr\bin\bash -lc "MSYSTEM= python3 run_tests.py --backend=ninja"
+      displayName: Run Tests
+    - task: PublishTestResults@2
+      inputs:
+        testResultsFiles: meson-test-run.xml
+        testRunTitle: 'Publish test results'
+        publishRunAttachments: true
```

Comment:
I think this also needs ` condition: not(canceled())` or similar, or this step isn't run  f the previous step fails (i.e. the test results aren't published if the tests failed), which isn't that useful.

I suggest it might be also worthwhile publishing the raw test logs as a build artefact.

Reply:
ok